### ISSUE TITLE
Build failure with CUDA 12

### ISSUE THREAD

**Issue by fvanmaele** [astra-toolbox/astra-toolbox#440]
Platform: Arch Linux
CUDA: 12.2.2
Nvidia driver version: 535.113.01
Host compiler: gcc 12

Trying to build ASTRA using the Linux instructions with CUDA 12 results in the following error:
```
/opt/cuda/bin/nvcc -Wno-deprecated-gpu-targets -gencode=arch=compute_50,code=sm_50 -gencode=arch=compute_60,code=sm_60 -gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_75,code=sm_75 -gencode=arch=compute_80,code=sm_80 -gencode=arch=compute_86,code=sm_86 -gencode=arch=compute_86,code=compute_86 -I/opt/cuda/include -I./../../include -DASTRA_CUDA -c ./../../cuda/2d/fan_fp.cu -Xcompiler -fPIC -DPIC -o cuda/2d/.libs/fan_fp.o
./../../cuda/2d/par_bp.cu(36): error: texture is not a template
  typedef texture<float, 2, cudaReadModeElementType> texture2D;
          ^

1 error detected in the compilation of "./../../cuda/2d/par_bp.cu".
make: *** [Makefile:376: cuda/2d/par_bp.lo] Error 2
make: *** Waiting for unfinished jobs....
```
The build works fine with CUDA 11.7

**Comment by askorikov** [comment]
Hi! I think this should be fixed by 490784100653a391166717ef32e652934d062564. Can you try pulling the latest changes?

**Comment by fvanmaele** [comment]
I can confirm a checkout from `master` builds successfully. Thanks!
